packages/block-editor/src/components/inserter/menu.js

```diff
@@ -344,9 +348,21 @@ export class InserterMenu extends Component {
 							</a>
 						</PanelBody>
 					) }
-					{ isEmpty( suggestedItems ) && isEmpty( reusableItems ) && isEmpty( itemsPerCategory ) && (
-						<p className="editor-inserter__no-results block-editor-inserter__no-results">{ __( 'No blocks found.' ) }</p>
-					) }
+					{
+						! isDownloadableBlocksEnabled && isMenuEmpty && (
```

Comment:
It feels like `@wordpress/block-editor` shouldn't have access to `isDownloadableBlocksEnabled`. If there needs to be different handling in case downloadable blocks are enabled, we can find a way to pass the default component when there are no fills registered. Although, I would personally seek a solution where this part is always rendered and fills take that into account.